packages/shared/src/multisig.proxy.ts

```diff
@@ -0,0 +1,1176 @@
+  const transferCall = client.api.tx.balances.transferKeepAlive(alice.address, transferAmount)
+
+  const announceTx = client.api.tx.proxy.announce(dave.address, transferCall.method.hash)
+
+  // 1. Alice creates a 2-of-3 multisig with Bob and Charlie.
+  const threshold = 2
+  const maxWeight = { refTime: 1000000000, proofSize: 1000000 } // Conservative weight limit
+  let otherSignatories = [bob.address, charlie.address].map((addr) => encodeAddress(addr, addressEncoding)).sort()
+
+  const announceMultiFirstTx = client.api.tx.multisig.asMulti(
+    threshold,
+    otherSignatories,
+    null, // No timepoint for first approval
+    announceTx.method.toHex(),
+    maxWeight,
+  )
+
+  let currBlockNumber = (await client.api.rpc.chain.getHeader()).number.toNumber()
+
+  await sendTransaction(announceMultiFirstTx.signAsync(alice))
+  await client.dev.newBlock()
+
+  // Check that Alice's multisig creation deposit was reserved.
+  const aliceReservedFunds = await getReservedFunds(client, alice.address)
+  expect(aliceReservedFunds, 'Alice should have reserved funds for multisig deposit').toBe(
+    await getMultisigCosts(client, threshold),
+  )
+
+  // Check the multisig creation event and extract the multisig account address.
+  const [multisigAddress, multisigExtrinsicIndex] = await getAndVerifyMultisigEventData(
+    client,
+    alice.address,
+    addressEncoding,
+  )
+
+  // Funds the multisig account in order to execute the call.
+  const multisigFunds = 100e10
+  await setupBalances(client, [{ address: multisigAddress, amount: multisigFunds }])
+
+  // 2. Dave nominates the multisig as his standard proxy.
+  const addProxyTx = client.api.tx.proxy.addProxy(multisigAddress, proxyType, 0)
+
+  await sendTransaction(addProxyTx.signAsync(dave))
+  await client.dev.newBlock()
+
+  // Check that Dave has had funds reserved for the proxy.
+  const daveReservedFunds = await getReservedFunds(client, dave.address)
+  expect(daveReservedFunds, 'Dave should have reserved funds').toBe(await getProxyCosts(client, 1))
+
+  // 3. Bob co-signs the multisig, which allows it to announce the transfer as a proxy of Dave.
+  otherSignatories = [alice.address, charlie.address].map((addr) => encodeAddress(addr, addressEncoding)).sort()
+  let announceMultiLastTx = client.api.tx.multisig.asMulti(
+    threshold,
+    otherSignatories,
+    {
+      height: currBlockNumber + 1,
+      index: multisigExtrinsicIndex,
+    },
+    announceTx.method.toHex(),
+    maxWeight,
+  )
+
+  const finalApprovalEvents = await sendTransaction(announceMultiLastTx.signAsync(bob))
+  await client.dev.newBlock()
+
+  await checkEvents(finalApprovalEvents, 'multisig')
+    .redact({
+      redactKeys: /height/,
+    })
+    .toMatchSnapshot('events when Bob approves multisig call')
+
+  const announcementObject = {
+    real: encodeAddress(dave.address, addressEncoding),
+    callHash: transferCall.method.hash.toHex(),
+    height: currBlockNumber + 3,
```

Comment:
Just a reminder that this needs to be general so as to accomodate chains in which the block provider is non-local.

Using the below function,

https://github.com/open-web3-stack/polkadot-ecosystem-tests/blob/dbb93f9a3b1e9fe413be03f226955295b5273207/packages/shared/src/helpers/index.ts#L456-L467

This becomes something like 

```suggestion
    const offset = blockProviderOffset(testConfig)
    height: currBlockNumber + 3 * offset,
```